src/FSharp.Core/list.fs

```diff
@@ -987,3 +988,174 @@ module List =
+        let inputLength = source.Length
+
+        if inputLength = 0 then
+            invalidArg "source" LanguagePrimitives.ErrorStrings.InputSequenceEmptyString
+
+        let i = Microsoft.FSharp.Primitives.Basics.Random.next randomizer 0 inputLength
+        source[i]
+
+    [<CompiledName("RandomChoice")>]
+    let randomChoice (source: 'T list) : 'T =
+        randomChoiceWith ThreadSafeRandom.Shared source
+
+    [<CompiledName("RandomChoicesWith")>]
+    let randomChoicesWith (random: Random) (count: int) (source: 'T list) : 'T list =
+        checkNonNull "random" random
+
+        if count < 0 then
+            invalidArgInputMustBeNonNegative "count" count
+
+        let inputLength = source.Length
+
+        if inputLength = 0 then
+            invalidArg "source" LanguagePrimitives.ErrorStrings.InputSequenceEmptyString
+
+        [
+            for _ = 0 to count - 1 do
+                let j = random.Next(0, inputLength)
+                source[j]
+        ]
+
+    [<CompiledName("RandomChoicesBy")>]
+    let randomChoicesBy (randomizer: unit -> float) (count: int) (source: 'T list) : 'T list =
+        if count < 0 then
+            invalidArgInputMustBeNonNegative "count" count
+
+        let inputLength = source.Length
+
+        if inputLength = 0 then
+            invalidArg "source" LanguagePrimitives.ErrorStrings.InputSequenceEmptyString
+
+        [
+            for _ = 0 to count - 1 do
+                let j = Microsoft.FSharp.Primitives.Basics.Random.next randomizer 0 inputLength
+                source[j]
+        ]
+
+    [<CompiledName("RandomChoices")>]
+    let randomChoices (count: int) (source: 'T list) : 'T list =
+        randomChoicesWith ThreadSafeRandom.Shared count source
+
+    [<CompiledName("RandomSampleWith")>]
+    let randomSampleWith (random: Random) (count: int) (source: 'T list) : 'T list =
+        checkNonNull "random" random
+
+        if count < 0 then
+            invalidArgInputMustBeNonNegative "count" count
+
+        let inputLength = source.Length
+
+        if inputLength = 0 then
+            invalidArg "source" LanguagePrimitives.ErrorStrings.InputSequenceEmptyString
+
+        if count >= inputLength then
+            invalidArgOutOfRange "count" count "source.Length" inputLength
+
+        // algorithm taken from https://github.com/python/cpython/blob/69b3e8ea569faabccd74036e3d0e5ec7c0c62a20/Lib/random.py#L363-L456
+        let setSize =
+            Microsoft.FSharp.Primitives.Basics.Random.getMaxSetSizeForSampling count
+
+        if inputLength <= setSize then
+            let pool = source |> toArray
+
+            [
+                for i = 0 to count - 1 do
+                    let j = random.Next(0, inputLength - i)
+                    let item = pool[j]
+                    pool[j] <- pool[inputLength - i - 1]
+                    item
+            ]
+        else
+            let selected = HashSet()
```

Comment:
Here and elsewhere:

```suggestion
            let selected = HashSet count
```

Reply:
Fixed

Reply:
@Lanayx It looks like you added `count` in a few places but not everywhere (including here).

Reply:
@brianrourkeboll I just had to revert anyway, since this constructor is unavailable in netstarndard2.0 https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.hashset-1.-ctor?view=net-8.0#system-collections-generic-hashset-1-ctor(system-int32)

Reply:
Ugh. The `netstandard2.0` limitation is so frustrating, especially in cases like this, where the need to support someone running .NET Framework 4.6.2 or 4.7.1 means that everyone else using _any runtime released since 2017_ can't benefit from BCL API improvements...